subprojects/build-init/src/main/groovy/org/gradle/buildinit/plugins/internal/maven/Maven2Gradle.groovy

```diff
@@ -187,10 +245,18 @@ ${globalExclusions(this.effectivePom)}
         exclusions
     }
 
-    def testNg = { moduleDependencies ->
+    def testNg(moduleDependencies) {
         if (moduleDependencies.contains('testng')) {
-            """test.useTestNG()
+            switch (scriptDsl) {
+                case KOTLIN:
+                    return """val test by tasks.getting(Test::class)
```

Comment:
Does `getting` not have a reified type version? Eg `getting<Test>()`?

Reply:
Nope, see https://github.com/gradle/kotlin-dsl/blob/6a3c51a65e8423aa33432b816996360cd32e7906/provider/src/main/kotlin/org/gradle/kotlin/dsl/NamedDomainObjectContainerExtensions.kt

Reply:
Can you open an issue to resolve this? Assign it to me and I'll take a crack at fixing it when I have a free moment.

Reply:
Would you please open that issue yourself on `kotlin-dsl`?

Reply:
Can do.

Reply:
Done: https://github.com/gradle/kotlin-dsl/issues/605

Reply:
:o: This code should be refactored so there's a single source of truth in the codebase for the right way of configuring tasks in Kotlin (or Groovy for that matter). The closest we have right now is `BuildScriptBuilder.PrettyPrinter` combined with `BuildScriptBuilder.KotlinSyntax`, it would make sense to refactor towards making them reusable here.

In any case, I'd like to suggest the `Maven2Gradle` changes to be moved to a separate PR so we can integrate the changes to `build-init` faster and later do the right thing with `Maven2Gradle`.